### ISSUE TITLE
L011: Aliasing preference for tables -  wrongly flagged as violation

### ISSUE THREAD

**Issue by dvd-p** [sqlfluff/sqlfluff#2723]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Rule L011 description: Implicit/explicit aliasing of table
Config set to `implicit`

When running `sqlfluff fix file_name.sql` it incorrectly flags a violation, and, if accepting the fix, it adds explicit `as`  in the table aliasing.

When running `sqlfluff fix file_name.sql` for a second time consecutively, it incorrectly flags a violation for L010 and, if accepting the fix, it turns the explicit `as` into `AS` 

When running `sqlfluff fix file_name.sql` for a third time consecutively, it correctly removes the table aliasing, which should not have been changed from the start.

### Expected Behaviour

Rule L011 is set to `implicit`, thus no violation should be flagged

### Observed Behaviour

Original query:

```
select
    foo as foo_alias
    , bar as bar_alias
from tbl_1 t1
    left join tbl_2 t2 on t1.foo = t2.foo;
```

1st `sqlfluff fix file_name.sql` run:
![20220227-Code-test3 sql](https://user-images.githubusercontent.com/46428930/155878047-151990db-b483-46d4-842e-9fdd6e2e66d8.png)

fix accepted:
```
select
    foo as foo_alias
    , bar as bar_alias
from tbl_1 as t1
    left join tbl_2 as t2 on t1.foo = t2.foo;
```

2nd `sqlfluff fix file_name.sql` run:
![20220227-Code-test3 sql](https://user-images.githubusercontent.com/46428930/155878130-b8820577-bc1a-420b-8d4b-1b472096c54a.png)

fix accepted:
```
select
    foo as foo_alias
    , bar as bar_alias
from tbl_1 AS t1
    left join tbl_2 AS t2 on t1.foo = t2.foo;
```

3rd: `sqlfluff fix file_name.sql` run:
![20220227-Code-test3 sql](https://user-images.githubusercontent.com/46428930/155878230-30bbdb4e-c6cc-4d03-ba4b-a1a2c9a3b755.png)

fix accepted:
```
select
    foo as foo_alias
    , bar as bar_alias
from tbl_1 t1
    left join tbl_2 t2 on t1.foo = t2.foo;
```

will go on a loop if continue running the fix command

### How to reproduce

running `sqlfluff fix file_name.sql` on the following query will reproduce the bug

```
select
    foo as foo_alias
    , bar as bar_alias
from tbl_1 t1
    left join tbl_2 t2 on t1.foo = t2.foo;
```

### Dialect

snowflake

### Version

sqlfluff, version 0.10.1

### Configuration

```
[sqlfluff:rules:L010]
# Keywords
capitalisation_policy = lower
# Comma separated list of words to ignore for this rule
ignore_words = None


[sqlfluff:rules:L011]
# Aliasing preference for tables
aliasing = implicit
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
I think this was fixed in #2683 which will be in the next release. Could you confirm it's fixed when using the latest `main` branch?

**Comment by dvd-p** [comment]
thanks @tunetheweb! yes, it's fixed with #2683. I searched beforehand for open tickets only

**Comment by RoyalTS** [comment]
@tunetheweb This has been tripping me up for days now. Any idea when the next release is going to be?

**Comment by tunetheweb** [comment]
This week! Already for the PR open in #2742 so just giving it a day or two for people to finish up open PRs.
